custom/ilsgateway/tasks.py

```diff
@@ -364,3 +368,67 @@ def third_soh_task():
     last_month_last_day, fifth_business_day = get_last_and_nth_business_day(now, 5)
     if now.day == fifth_business_day.day:
         send_for_all_domains(last_month_last_day, SOHReminder)
+
+
+def recalculate_on_group_change(location, last_run):
+    OrganizationSummary.objects.filter(location_id=location.get_id).delete()
+    process_facility_warehouse_data(location, default_start_date(), last_run.end)
+
+    for parent in location.sql_location.get_ancestors(ascending=True):
+        process_non_facility_warehouse_data(parent.couch_location,
+                                            default_start_date(), last_run.end, strict=False)
+
+
+def recalculate_on_parent_change(location, previous_parent_id, last_run):
+    previous_parent = SQLLocation.objects.get(location_id=previous_parent_id)
+    type_location_map = defaultdict(set)
+
+    previous_ancestors = list(previous_parent.get_ancestors(include_self=True, ascending=True))
+    actual_ancestors = list(location.sql_location.get_ancestors(ascending=True))
```

Comment:
From doc: https://django-mptt.github.io/django-mptt/models.html#get-ancestors-ascending-false-include-self-false. "passing True for the ascending argument will reverse the ordering (immediate parent first, root ancestor last)."